win32/src/gballoc_ll_jemalloc.c

```diff
@@ -214,3 +218,97 @@ void gballoc_ll_print_stats(void)
     /* Codes_SRS_GBALLOC_LL_JEMALLOC_01_008: [ gballoc_ll_print_stats shall call je_malloc_stats_print and pass to it jemalloc_print_stats_callback as print callback. ]*/
     je_malloc_stats_print(jemalloc_print_stats_callback, NULL, NULL);
 }
+
+int gballoc_ll_set_decay(int64_t decay_milliseconds)
+{
+    int result = 0;
+
+    /* Codes_SRS_GBALLOC_LL_JEMALLOC_28_001: [ gballoc_ll_set_decay shall advance jemalloc's internal epoch counter. ]*/
+    if (je_mallctl("epoch", NULL, NULL, NULL, 0) != 0)
+    {
+        /* Codes_SRS_GBALLOC_LL_JEMALLOC_28_008: [ If there are any errors, gballoc_ll_set_decay shall fail and return a non-zero value. ]*/
+        LogError("failed to advance jemalloc epoch");
+        result = MU_FAILURE;
+    }
+    else
+    {
+        /* Codes_SRS_GBALLOC_LL_JEMALLOC_28_002: [ gballoc_ll_set_decay shall set the dirty decay time for new arenas to decay_milliseconds milliseconds. ]*/
+        if (je_mallctl("arenas.dirty_decay_ms", NULL, NULL, &decay_milliseconds, sizeof(decay_milliseconds)) != 0)
+        {
+            /* Codes_SRS_GBALLOC_LL_JEMALLOC_28_008: [ If there are any errors, gballoc_ll_set_decay shall fail and return a non-zero value. ]*/
+            LogError("failed to set dirty decay time");
+            result = MU_FAILURE;
+        }
+        else
+        {
+            /* Codes_SRS_GBALLOC_LL_JEMALLOC_28_003: [ gballoc_ll_set_decay shall set the muzzy decay time for new arenas to decay_milliseconds milliseconds. ]*/
+            if (je_mallctl("arenas.muzzy_decay_ms", NULL, NULL, &decay_milliseconds, sizeof(decay_milliseconds)) != 0)
+            {
+                /* Codes_SRS_GBALLOC_LL_JEMALLOC_28_008: [ If there are any errors, gballoc_ll_set_decay shall fail and return a non-zero value. ]*/
+                LogError("failed to set muzzy decay time"); 
+                result = MU_FAILURE; 
+            }
+            else
+            {
+                uint64_t narenas;
```

Comment:
>narenas [](http://example.com/codeflow?start=25&length=7)

according to documentation, "narenas" is "unsigned".

Code shoudl assert that sizeof(unsigned) is the same as sizeof("uint64_t).

Because what if they aren't.

Documentation here: `[https://jemalloc.net/jemalloc.3.html`](https://jemalloc.net/jemalloc.3.html`)

Reply:
The unsigned size would be platform dependent. So the static assert might fail if the machine's not 64-bit. Do you think using unsigned directly in the code would be a good idea?

Reply:
I do not believe using "unsigned" directly in the code would be a good idea.

Reply:
As discussed offline, I have changed it to uin32_t and added a static assert.